### PR TITLE
feat: convert json to interfaces

### LINKED ISSUE #3551: [feature]: Ability to convert JSON responses into programming language interface/types
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Would be great to have an option to export/copy JSON responses as interface/type in the programming language of choice like ThunderClient does or this website [quicktype.io](https://quicktype.io/)

### Why should this be worked on?

Some other REST Clients implement this feature an it is very useful in some cases where you are unable to get to the response interface

### PR DESCRIPTION
**Pull request by rajdip-b**
Closes #3551 

### Description
Added a feature to generate the interface type of JSON response into any other language.
image

### Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### Additional Information
Current behaviour:
image
image

Generated TypeScript interface:
image

### CODE REVIEW

**Comment by rajdip-b** [comment]
@AndrewBastin hello, can i get a review on this implementation?

File: packages/hoppscotch-common/package.json
Comment (AndrewBastin) [inline packages/hoppscotch-common/package.json:79]: Since you have added a dependency, you would have had an updated `pnpm-lock.yaml` file. Please commit that in as well since we require lockfiles to be up-to-date with dependencies.

**Review by AndrewBastin — CHANGES_REQUESTED** [review]

File: packages/hoppscotch-common/package.json
Comment (rajdip-b) [reply packages/hoppscotch-common/package.json:79]: Sure thing!

Comment (rajdip-b) [reply packages/hoppscotch-common/package.json:79]: @AndrewBastin done. Can you review now?

**Comment by anwarulislam** [comment]
Fix this design inconsistency: The sticky action bar should be 34px


image

Ref:
image

**Comment by anwarulislam** [comment]
Use existing select/option style:

image


Ref:
image

**Comment by rajdip-b** [comment]
> Fix this design inconsistency: The sticky action bar should be 34px
> image
> 
> Ref: image

Hey @anwarulislam, I did notice this inconsistency before. The only reason I let it be was because if I didn't put some spacing in the area, the dropdown button was getting squashed. Do you still want me to go ahead?

**Comment by rajdip-b** [comment]
> Use existing select/option style:
> 
> image
> Ref: image

Okay I'll switch to this one. Should the color of the button be the theme black or theme purple?

**Review by anwarulislam — CHANGES_REQUESTED** [review]
- TypeScript should be default as we already able to copy JSON by a different button

**Comment by rajdip-b** [comment]
> * TypeScript should be default as we already able to copy JSON by a different button

Actually it's cJSON, it was not at all similar to JSON I did check that. Anyway, I'll change the default to TypeScript!

**Comment by anwarulislam** [comment]
> Hey @anwarulislam, I did notice this inconsistency before. The only reason I let it be was because if I didn't put some spacing in the area, the dropdown button was getting squashed. Do you still want me to go ahead?

Can you think of a different approach in terms of this sticky action bar UI?

**Comment by anwarulislam** [comment]
> > > Hey @anwarulislam, I did notice this inconsistency before. The only reason I let it be was because if I didn't put some spacing in the area, the dropdown button was getting squashed. Do you still want me to go ahead?
> > 
> > 
> > Can you think of a different approach in terms of this sticky action bar UI?
> 
> Maybe we can move the button out of the action bar itself, place it to the right of the action bar?

@liyasthomas can you help him with this UI part?

**Comment by liyasthomas** [comment]
Sure. I'll get back to this in few hours.

**Comment by rajdip-b** [comment]
@anwarulislam @liyasthomas hey folks, waiting on the design confirmation.

**Comment by liyasthomas** [comment]
@rajdip-b thanks for the contributions. I'll take over minor design changes to avoid the layout shifts happening by selecting an interface language from the drop-down. Please allow me a few hours to get back to this.

**Comment by rajdip-b** [comment]
Sure thing!!

**Comment by liyasthomas** [comment]
I've ported the feature to the GraphQL playground as well.

**Review by liyasthomas — APPROVED** [review]
LGTM 💯

**Comment by rajdip-b** [comment]
@liyasthomas thanks for approving the feature! Is there anything more that you want me to add? Or would you be making the changes that @anwarulislam suggested?

**Comment by liyasthomas** [comment]
I already fixed the margin issue, refer here: https://github.com/hoppscotch/hoppscotch/pull/3566#issuecomment-1833803523

Hence I don't think this PR need any changes. Anyways, I'll wait till we get another review.

**Review by nivedin — APPROVED** [review]
I have updated the tippy title and toast message to give user better feedback.
LGTM 💯

**Review by liyasthomas — APPROVED** [review]
LGTM

**Pull request merged**
